Index: sub_range.hpp
===================================================================
--- sub_range.hpp	(revision 39713)
+++ sub_range.hpp	(working copy)
@@ -88,16 +88,18 @@
             return *this;
         }
 
-        sub_range& operator=( sub_range r )
+#if BOOST_WORKAROUND(BOOST_MSVC, == 1310) || BOOST_WORKAROUND(BOOST_MSVC, == 1400)
+        sub_range( const sub_range& r ) :
+            base( static_cast<const base&>( r ) )
+        { }
+
+        sub_range& operator=( const sub_range& r )
         {
-            //
-            // argument passed by value to avoid 
-            // const_iterator to iterator conversion
-            //
-            base::operator=( r );
-            return *this;            
+            base::operator=( static_cast<const base&>( r ) );
+            return *this;
         }
-        
+#endif
+
     public:
         
         iterator        begin()          { return base::begin(); }
