Index: transform_iterator.hpp
===================================================================
--- transform_iterator.hpp	(revision 41962)
+++ transform_iterator.hpp	(working copy)
@@ -80,7 +80,7 @@
             transform_iterator<UnaryFunc, Iterator, Reference, Value>
           , Iterator
           , cv_value_type
-          , use_default    // Leave the traversal category alone
+          , typename pure_traversal_tag<typename iterator_traversal<Iterator>::type>::type
           , reference
         > type;
     };
